core/statedb/state_cache.go

```diff
@@ -110,6 +114,13 @@ func (c *StateCache) SetPendingUpdateAccount(accountIndex int64, account *types.
 	c.PendingUpdateAccountMap[accountIndex] = account
 }
 
+func (c *StateCache) SetPendingUpdateGasAccount(account *types.AccountInfo, assetId int64, balanceDelta *big.Int) {
+	if c.PendingGasAccount == nil {
+		c.PendingGasAccount = account.DeepCopy()
```

Comment:
Only DeepCopy is not enough. `GetPeingAccount` will return two items for same `GasAccount`, and in block commit process, the second item will be used which override first item

Reply:
fixed